pkg/controller/volume/attachdetach/util/util.go

```diff
@@ -159,6 +159,23 @@ func getPVSpecFromCache(name string, pvcReadOnly bool, expectedClaimUID types.UI
 	return volume.NewSpecFromPersistentVolume(clonedPV, pvcReadOnly), nil
 }
 
+// DetermineVolumeAction returns true if volume and pod needs to be added to dswp
+// and it returns false if volume and pod needs to be removed from dswp
+func DetermineVolumeAction(pod *v1.Pod, desiredStateOfWorld cache.DesiredStateOfWorld, defaultValue bool) bool {
+	if pod == nil || len(pod.Spec.Volumes) <= 0 {
+		return defaultValue
```

Comment:
This implies this function could return `true` if pod is nil or has no volumes... Is this intended?